AxonDeepSeg/apply_model.py

```diff
@@ -1,73 +1,166 @@
+    else:
+        # just return the first .pth file found
+        next((checkpoint_folder_path).glob('*.pth')).name
+
+
+def extract_from_nnunet_prediction(pred, pred_path, class_name, class_value) -> str:
     '''
+    Extracts the given class from the nnunet raw prediction, saves it in a 
+    separate mask and return the path of the extracted mask.
 
-    path_model=path_model_folder
-    input_filenames = acquisitions_filenames
+    Parameters
+    ----------
+    pred : np.ndarray
+        The raw prediction from nnunet with values 0, 1, 2, ...
+    pred_path : pathlib.Path
+        Path to the raw prediction file; We expect its filename to end 
+        with '_seg-nnunet.png'
+    class_name : str
+        Name of the class to extract. e.g. 'axon', 'myelin', etc.
+    class_value : int
+        Value of the class in the raw prediction.
 
-    # Fill options dictionary
-    options = {"pixel_size": [acquired_resolution, acquired_resolution], "pixel_size_units": "um", "binarize_maxpooling": True}
-    if no_patch:
-        options["no_patch"] = no_patch
-        logger.warning("The 'no-patch' option was selected for segmentation. "\
-                       "Please note that it may not be suitable with large images depending on computer RAM capacity.")
-    else:
-        logger.warning("The 'no-patch' option was not selected for segmentation. "\
-                       "Please note that this option could potentially produce better results but may not be suitable "\
-                       "with large images depending on computer RAM capacity.")
-    if overlap_value:
-        # When both no_patch and overlap_value are used, the no_patch option supersedes the overlap_value
-        # and a warning will be issued by ivadomed while segmenting without patches.
-        options["overlap_2D"] = overlap_value
-    elif not no_patch:
-        # Default overlap is used only without the no_patch option.
-        options["overlap_2D"] = [default_overlap, default_overlap]
-
-    # IVADOMED automated segmentation
-    nii_lst, _ = imed_inference.segment_volume(str(path_model), input_filenames, gpu_id=gpu_id, options=options)
-    
-    target_lst = [str(axon_suffix), str(myelin_suffix)]
+    Errors
+    ------
+    ValueError
+        If the class value is not found in the raw prediction.
+    ValueError
+        If the raw nnunet prediction file does not end with '_seg-nnunet.png'.
 
-    imed_inference.pred_to_png(nii_lst, target_lst, str(Path(input_filenames[0]).parent / Path(input_filenames[0]).stem))
-    if verbosity_level >= 1:
-        print(Path(path_acquisitions_folders) / (Path(input_filenames[0]).stem + str(axonmyelin_suffix)))
+    Returns
+    -------
+    new_fname : str
+        Path to the extracted class mask saved.
+    '''
+
+    pred_path = ads_utils.convert_path(pred_path)
+    if not np.any(pred == class_value):
+        raise ValueError(f'Class value {class_value} not found in the raw prediction.')
```

Comment:
self: line not covered by tests

Reply:
Covered in [d479b6c](https://github.com/axondeepseg/axondeepseg/pull/800/commits/d479b6cdf7843d4394afb17bad6de2395d94a05e)